src/compiler/checker.ts

```diff
@@ -26774,6 +26781,12 @@ namespace ts {
             diagnostics.add(resultDiagnostic);
         }
 
+        function containerSeemstoBeEmptyDomElement(containingType: Type) {
+            return (compilerOptions.lib && !compilerOptions.lib.includes("dom")) &&
+                everyContainedType(containingType, type => type.symbol && /^(EventTarget|Node|((HTML[a-zA-Z]*)?Element))$/.test(type.symbol.escapedName.toString())) &&

```

Comment:
@sandersn was there a reason you asked to not use a regexp in https://github.com/microsoft/TypeScript/pull/34407#discussion_r389982833? HTML*Element seems easiest to check that way.

I'm also not clear on whether `isEmptyObjectType` from #34407 or the `getPropertiesOfType` approach here would be preferred. Do you have guidance?

Reply:
The earlier regex was complicated enough that I couldn't easily imagine what it would accept. This one is a lot simpler, so I think it's OK.

`isEmptyObjectType` is super comprehensive and also easier to read, so I'd use that.